cmd/scollector/collectors/continuous.go

```diff
@@ -0,0 +1,95 @@
+package collectors
+
+import (
+	"reflect"
+	"regexp"
+	"runtime"
+	"time"
+
+	"bosun.org/collect"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+)
+
+type ContinuousCollector struct {
+	F    func(collectorStatsChan chan<- *ContinuousCollectorStats)
+	name string
+	tags opentsdb.TagSet
+	init func()
+}
+
+type ContinuousCollectorStats struct {
+	duration float64
+	result   int
+}
+
+var ContinuousCollectorVars struct {
+	dpChan                chan<- *opentsdb.DataPoint
+	quit                  <-chan struct{}
+	reTwoOrMoreUnderscore *regexp.Regexp
+	nameWorkerChan        map[string]chan *ContinuousCollectorStats
+}
+
+func init() {
+	ContinuousCollectorVars.reTwoOrMoreUnderscore = regexp.MustCompile("[_]{2,}")
+	ContinuousCollectorVars.nameWorkerChan = make(map[string]chan *ContinuousCollectorStats)
+}
+
+func (c *ContinuousCollector) Init() {
+	if c.init != nil {
+		c.init()
+	}
+}
+
+func (c *ContinuousCollector) Run(dpChan chan<- *opentsdb.DataPoint, quit <-chan struct{}) {
```

Comment:
You don't have to do this all in one collector. You can start one interval collector per query and they will all run independently, and slow queries won't affect fast ones.